client/src/search.tsx

```diff
@@ -13,468 +14,89 @@ function isMobileUserAgent() {
-      onSubmit={(e) => {
-        e.preventDefault();
-      }}
-    >
-      <SearchWidget
-        inputRef={inputRef}
-        pathname={pathname}
-        value={query}
-        onChange={(value) => setQuery(value)}
-        onSelect={(url) => {
-          navigate(url);
-          setQuery("");
-        }}
-      />
-    </form>
-  );
-}
-
-const ACTIVE_PLACEHOLDER = "Go ahead. Type your search...";
 const INITIALIZING_PLACEHOLDER = "Initializing search...";
+const ACTIVE_PLACEHOLDER = "Go ahead. Type your search...";
 // Make this one depend on figuring out if you're on a mobile device
 // because there you can't really benefit from keyboard shortcuts.
 const INACTIVE_PLACEHOLDER = isMobileUserAgent()
   ? "Site search..."
   : 'Site search... (Press "/" to focus)';
 
-// TODO the only reason exporting this, for now, is to make
-// jest tests pass until https://github.com/mdn/yari/pull/494
-// is resolved.
-export class SearchWidget extends React.Component<{
-  value: string;
-  pathname?: string;
-  onChange: (url: string) => unknown;
-  inputRef: React.RefObject<HTMLInputElement>;
-  onSelect?: (url: string) => unknown;
-  onValueExistsChange?: (exists: boolean) => unknown;
-}> {
-  static defaultProps = {
-    inputRef: React.createRef(null),
-  };
-
-  state = {
-    highlitResult: null,
-    initialized: null, // null=not started, false=started, true=finished
-    searchResults: [],
-    serverError: null,
-    showSearchResults: false,
-  };
-
-  getCurrentLocale = () => {
-    return (
-      (this.props.pathname && this.props.pathname.split("/")[1]) || "en-US"
-    );
+type Titles = {
+  [url: string]: {
+    title: string;
+    popularity: number;
   };
-
-  componentDidUpdate(prevProps) {
-    if (prevProps.pathname !== this.props.pathname) {
-      // Hide search results if you changed page.
-      if (this.state.showSearchResults || this.state.q) {
-        this.setState({
-          highlitResult: null,
-          q: "",
-          showSearchResults: false,
-          locale: this.props.pathname.split("/")[1] || "en-US",
-        });
+};
+
+type SearchIndex = {
+  flex: any;
+  fuzzy: FuzzySearch;
+  titles: Titles;
+};
+
+function useSearchIndex(): [null | SearchIndex, () => void] {
+  const [shouldInitialize, setShouldInitialize] = useState(false);
+  const [searchIndex, setSearchIndex] = useState<null | SearchIndex>(null);
+  const locale = useParams().locale || "en-US";
+
+  const url = `/${locale}/titles.json`;
+  const { error, data: fetchedTitles } = useSWR<Titles>(
+    shouldInitialize ? url : null,
+    async (url) => {
+      const response = await fetch(url);
+      if (!response.ok) {
+        throw new Error(await response.text());
       }
-    }
-
-    if (prevProps.value !== this.props.value) {
-      this.updateSearch();
-    }
-  }
+      const { titles } = await response.json();
+      localStorage.setItem(url, JSON.stringify(titles));
+      return titles;
+    },
+    { revalidateOnFocus: false }
```

Comment:
By the way, one idea I'm starting to love is to use:
```suggestion
    { revalidateOnFocus: process.env.NODE_ENV === 'development' }
```
because when you're doing local development, it's very possible that the `/$locale/titles.json` URL *has* changed. Whereas in prod, that almost never happens. 

Or, in the case of an app I was working on for a side-project, by having that trick, it meant that every time I switched from the browser to my editor (to work on the `thing/views.py` code), it would refresh the search to my backend immediately when I changed focus back on the browser. Neat!

Not sure if it applies here. Just a note really.